cassdegrees/ui/forms.py

```diff
@@ -89,6 +108,45 @@ def clean_units(self):
             raise forms.ValidationError("Units should be a multiple of 6!")
         return data
 
+    # Override clean to return links to existing content if unique_together constraint fails
+    # https://stackoverflow.com/questions/4659360/get-django-object-id-based-on-model-attribute
+    def clean(self):
+        cleaned_data = super().clean()
+
+        # check assignment as keys may not exist in cleaned dictionary if field level validation has failed
+        try:
+            draft_code = cleaned_data['code']
```

Comment:
Is there no way to hook into the pre-existing error checking here? No biggie if not.

Reply:
We're hooking into the pre-existing clean method. But the pre-existing validate_unique() method is a nightmare and goes several layers down (multiple overrides) and from what I can see has no easy place to hook in to actually return a conflicting id.

This method has the benefit that the natural errors raised by validate_unique will specify the conflicting fields with the simple override we've already put in the NON_FIELD_ERRORS, so we can make some generic 'get conflicting record link' methods.